File: go/benchmark/benchmark.py
def criar_matriz(linhas, colunas):
    contador = 0
    matriz = []
    for i in range(linhas):
        linha = []
        for j in range(colunas):
            linha.append(contador)
            contador += 1
        matriz.append(linha)
    return matriz

def multiplicar_matriz(matriz, fator):
    nova_matriz = []
    for linha in matriz:
        nova_linha = []
        for elemento in linha:
            novo_valor = elemento * fator
            nova_linha.append(novo_valor)
        nova_matriz.append(nova_linha)
    return nova_matriz

# Uso
original = criar_matriz(2, 2)
multiplicada = multiplicar_matriz(original, 42)

File: go/benchmark/benchmark.go
package main

func criarMatriz(linhas, colunas int) [][]int {
	matriz := make([][]int, linhas)
	contador := 0

	for i := 0; i < linhas; i++ {
		matriz[i] = make([]int, colunas)
		for j := 0; j < colunas; j++ {
			matriz[i][j] = contador
			contador++
		}
	}
	return matriz
}

func multiplicarMatriz(matriz [][]int, fator int) [][]int {
	linhas := len(matriz)
	colunas := len(matriz[0])
	novaMatriz := make([][]int, linhas)

	for i := 0; i < linhas; i++ {
		novaMatriz[i] = make([]int, colunas)
		for j := 0; j < colunas; j++ {
			novaMatriz[i][j] = matriz[i][j] * fator
		}
	}
	return novaMatriz
}

func main() {
	original := criarMatriz(21000, 21000) //
	multiplicada := multiplicarMatriz(original, 42)
	_ = multiplicada[0][0]
}
